DECISION_LOG.md

```diff
@@ -0,0 +1,30 @@
+# Decision Log
+
+This document contains a log of major decisions made during the development of this project.
+A decisions provides context on the project and the reasons behind certain choices.
+
+## Decisions
+The most recent decision should be at the top.
+
+### 2025-01-21 Should we use Helm Charts?
+
+- more complex
```

Comment:
I would add that it allows having logic in templates, which makes them self-sufficient to figure out what to deploy into each environment, and do not spread that logic across CI/CD tool and template engine.

Reply:
```suggestion
Helm charts provides templating, which provides a flexible way to deploy into multiple environment based on values from that environment. This greatly reduces the spread that logic across CI/CD tools.

- more complex
```
something like this? feel free to make code suggesions